src/main/java/org/broadinstitute/hellbender/tools/exome/SegmentUtils.java

```diff
@@ -13,292 +15,452 @@
-                (columns, formatExceptionFactory) -> {
-                    // Note that Segment_Call is optional
-                    if (!columns.containsAll(SegmentTableColumns.SAMPLE.toString(), SegmentTableColumns.CONTIG.toString(),
-                            SegmentTableColumns.START.toString(), SegmentTableColumns.END.toString(), SegmentTableColumns.NUM_PROBES.toString(),
-                            SegmentTableColumns.MEAN.toString())) {
-                        throw formatExceptionFactory.apply("Bad header");
-                    }
-                    // return the lambda to translate dataLines into called segments.
-                    return dataLineToModeledSegmentFunction;
-                })) {
-            return reader.stream().collect(Collectors.toList());
-        } catch (final IOException | UncheckedIOException e) {
-            throw new UserException.CouldNotReadInputFile(segmentsFile, e);
-        }
-    }
+    /*===============================================================================================================*
+     * PUBLIC METHODS FOR SEGMENT UNION (COMBINING TARGET AND SNP SEGMENTS)                                          *
+     *===============================================================================================================*/
 
     /**
-     * the mean of all overlapping targets' coverages
-     *
-     * @throws IllegalStateException if overlapping targets have not been assigned or if no overlapping targets were found.
+     * Target segments from CBS (i.e., {@link PerformSegmentation}) are specified by target-end--target-end intervals.
+     * This method converts these to the corresponding target-start--target-end intervals, returning a new,
+     * modifiable list of segments.
      */
-    public static double meanTargetCoverage(final Locatable loc, final TargetCollection<TargetCoverage> targets) {
-        Utils.nonNull(loc, "Can't get mean coverage of null segment.");
-        Utils.nonNull(targets, "Mean target coverage requires non-null targets collection.");
+    public static List<SimpleInterval> fixTargetSegmentStarts(final List<SimpleInterval> targetSegments,
+                                                              final TargetCollection<? extends Locatable> targets) {
+        Utils.nonNull(targetSegments, "The list of target segments cannot be null.");
+        Utils.nonNull(targets, "The collection of targets cannot be null.");
 
-        final List<TargetCoverage> myTargets = targets.targets(loc);
-
-        if (myTargets.size() == 0) {
-            throw new IllegalStateException("Empty segment -- no overlapping targets.");
+        final List<SimpleInterval> targetSegmentsFixed = new ArrayList<>();
+        for (final SimpleInterval segment : targetSegments) {
+            try {
+                final Locatable firstTarget = targets.targets(segment).get(0);
+                if (firstTarget.getEnd() == segment.getStart()) {
```

Comment:
hmm, i guess i should also add an else throw exception, just in case there are targets in the segment but the segment start doesn't correspond to the first target.  good catch!